components/api_calls/build_calls.js

```diff
@@ -6,14 +6,18 @@ export async function getBuilds(searchParams) {
     for (const key in searchParams) {
         let value = searchParams[key];
 
-        if ((value[0] === "\"" && value[value.length - 1] === "\"") || (value[0] === "'" && value[value.length - 1] === "'")) {
-            value = value.substring(1, value.length - 1);
+        // Check if the value is enclosed in double or single quotes.
+        const isEnclosedInQuotes = (value[0] === "\"" && value[value.length - 1] === "\"") || (value[0] === "'" && value[value.length - 1] === "'");
+        
+        if (isEnclosedInQuotes) {
+            value = value.substring(1, value.length - 1); // Strip the quotes
+            params[key] = value; // Exact match
+        } else {
+            if (key !== "time_iso" && key !== "page" && key !== "brew_task_state")
+                params[`${key}__icontains`] = value; // Partial match
+            else
+                params[key] = value; // Exact match for the excepted keys
```

Comment:
So basically this is saying exact match for `time_iso`, `page` and `brew_task_state` right? So isn't a better way to say:

```
if (key === "time_iso" && key === "page" && key === "brew_task_state")
    params[key] = value;
else
    params[`${key}__icontains`] = value;
```
which is easier to understand. Unless you do not think there is no better way than this? :wink: _(Problems of using double negatives)_

Reply:
The variable key can hold only one value at a time. In the statement:
`if (key === "time_iso" && key === "page" && key === "brew_task_state")`
The && operator (logical AND) checks if all conditions are true. For the entire statement to be true, key would have to be equal to "time_iso", "page", and "brew_task_state" at the same time, which is not possible.

Instead, using `OR ('||')` will correctly check if 'key' matches any of the three values.
I've made the necessary adjustment. Thanks for pointing it out!

Reply:
yeah my bad, I wasn't looking at the logical operators. Good catch.